Content.Client/Viewport/GameScreenBase.cs

```diff
@@ -83,6 +89,14 @@ public override void FrameUpdate(FrameEventArgs e)
                     renderScale = svp.CurrentRenderScale;
                 }
             }
+            else if (UserInterfaceManager.CurrentlyHovered is EntityMenuElement element)
+            {
+                entityToClick = element.Entity;
+
+                // BUG: This assumes that the main viewport is the viewport that the context menu is active on. This
+                // might be particularly important in the future with a more advanced mapping mode.
+                renderScale = _eyeManager.MainViewport.GetRenderScale();
```

Comment:
this is incorrect even now i think because of chess and what not, though you might not be able to use the context menu there anyway. is there not a CurrentViewport or anything like that

Reply:
>  might not be able to use the context menu there anyway

Because the chess entities are in a separate map, the in-range-unobstructed checks fail. You can still right click them if you use `/toggleallcontext` (I think it would also work for ghosts?).

>  is there not a CurrentViewport or anything like that

Well, a bit above the code you highlighted it uses `UserInterfaceManager.CurrentlyHovered is IViewportControl vp` to get the viewport, but in this specific case we are hovering over a context menu which could be a popup above several other viewports, and the entity in that menu could be present in more than one viewport.

I just copied this code & the "BUG" disclaimer over, it was previously in some other context menu file. At the time I didn't really think too much about it, but my conclusion now is that the hover-highlight shader just needs to be re-done to properly support multiple viewports. It probably shouldn't need to get the currently used viewport / the current render scale at all. Ignoring the entity context menu aside, the way it currently works is already broken.

When hovering over an entity, it updates the `InteractionOutlineComponent` with the renderscale for the currently hovered viewport, which completely ignores the fact that this entity may be visible in more than one viewport simultaneously. For example. this is what you get when you have two windows open for the same chess game, with different viewport scaling:

Outline scale set to zoomed-out chess viewport:
![hover_1](https://user-images.githubusercontent.com/60421075/136149630-5882be1a-3db7-41dc-9ea1-f44775357247.png)
Outline scale set to zoomed-in chess viewport
![hover_2](https://user-images.githubusercontent.com/60421075/136149697-68e3197a-03eb-4d4d-84da-f542740a6651.png)
 Outline scale based on context menu (currently `MainViewport`, which corresponds to neither of the previous two):
![hover_3](https://user-images.githubusercontent.com/60421075/136149771-d404c427-61e1-43b1-aa72-f62167809cf8.png)

Reply:
sounds alright then to keep it how it is